core/ArchiveProcessor/PluginsArchiver.php

```diff
@@ -177,15 +177,7 @@ public function callAggregateAllPlugins($visits, $visitsConverted, $forceArchivi
                         $this->params->getSegment() ? sprintf("(for segment = '%s')", $this->params->getSegment()->getString()) : ''
                     );
                 } catch (Exception $e) {
-                    $className = get_class($e);
-
-                    if ($className === 'PHPUnit_Framework_Exception' || (class_exists('PHPUnit_Framework_Exception', false) &&  is_subclass_of($className, 'PHPUnit_Framework_Exception'))) {
-                        $exception = new $className($e->getMessage() . " - caused by plugin $pluginName", $e->getCode(), $e->getFile(), $e->getLine(), $e);
-                    } else {
-                        $exception = new $className($e->getMessage() . " - caused by plugin $pluginName", $e->getCode(), $e);
-                    }
-
-                    throw $exception;
+                    throw new PluginsArchiverException($e->getMessage() . " - in plugin $pluginName", $e->getCode(), $e);
```

Comment:
I suppose that should work. Although wondering if anything may trigger like an `NoAccessException` which should maybe show login page etc. I don't think it's an actual issue though but maybe could be. Not sure if we handle some other exceptions in any specific way.

Reply:
we could use reflection to change the `$message` property of the exception. Not sure how future proof that is, but I suppose the tests will tell us quickly if it breaks.

Reply:
Could work as well I suppose. I reckon it'd be just like https://stackoverflow.com/a/29542873 

Although I reckon the current approach is fine as well, just not sure if any specific exceptions would be treated differently.

Reply:
@tsteur Does that mean this PR is ok to merge or do you want to think about it some more?

Reply:
I reckon it should be fine 👍